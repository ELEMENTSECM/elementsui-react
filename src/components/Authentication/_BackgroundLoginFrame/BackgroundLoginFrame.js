import * as React from 'react';
import PropTypes from 'prop-types';

function BackgroundLoginFrame({ source }) {
	return (
		<div style={{ display: 'none' }}>
			<iframe
				src={source}
				id="background-login"
				style={{ display: 'none' }}
				tabIndex="-1"
				height="0"
				width="0"
				title="Empty"
			/>
			<iframe id="manual-login" tabIndex="1" height="500" width="500" title="Empty" />
		</div>
	);
}

BackgroundLoginFrame.propTypes = {
	/** BackgroundLoginFrame source */
	source: PropTypes.string
};

export default BackgroundLoginFrame;
