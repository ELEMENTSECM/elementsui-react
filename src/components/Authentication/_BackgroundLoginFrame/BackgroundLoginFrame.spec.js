import * as React from 'react';
import { shallow } from 'enzyme';
import BackgroundLoginFrame from './index';

describe('BackgroundLoginFrame', () => {
	test('displays correctly', () => {
		const wrapper = shallow(<BackgroundLoginFrame source="http://localhost" />);
		expect(wrapper).toMatchSnapshot();
	});
});
