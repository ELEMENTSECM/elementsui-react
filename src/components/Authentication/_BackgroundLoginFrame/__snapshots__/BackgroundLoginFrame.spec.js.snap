// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundLoginFrame displays correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BackgroundLoginFrame
    source="http://localhost"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <iframe
          height="0"
          id="background-login"
          src="http://localhost"
          style={
            Object {
              "display": "none",
            }
          }
          tabIndex="-1"
          title="Empty"
          width="0"
        />,
        <iframe
          height="500"
          id="manual-login"
          tabIndex="1"
          title="Empty"
          width="500"
        />,
      ],
      "style": Object {
        "display": "none",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "0",
          "id": "background-login",
          "src": "http://localhost",
          "style": Object {
            "display": "none",
          },
          "tabIndex": "-1",
          "title": "Empty",
          "width": "0",
        },
        "ref": null,
        "rendered": null,
        "type": "iframe",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "height": "500",
          "id": "manual-login",
          "tabIndex": "1",
          "title": "Empty",
          "width": "500",
        },
        "ref": null,
        "rendered": null,
        "type": "iframe",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <iframe
            height="0"
            id="background-login"
            src="http://localhost"
            style={
              Object {
                "display": "none",
              }
            }
            tabIndex="-1"
            title="Empty"
            width="0"
          />,
          <iframe
            height="500"
            id="manual-login"
            tabIndex="1"
            title="Empty"
            width="500"
          />,
        ],
        "style": Object {
          "display": "none",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "0",
            "id": "background-login",
            "src": "http://localhost",
            "style": Object {
              "display": "none",
            },
            "tabIndex": "-1",
            "title": "Empty",
            "width": "0",
          },
          "ref": null,
          "rendered": null,
          "type": "iframe",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": "500",
            "id": "manual-login",
            "tabIndex": "1",
            "title": "Empty",
            "width": "500",
          },
          "ref": null,
          "rendered": null,
          "type": "iframe",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
